### PR TITLE
solve issues 94...97; 99; 101; 102

### PR DESCRIPTION
**Pull request by berndfinger**

### CODE REVIEW

File: tasks/RedHat/installation.yml [hunk 1]
```diff
@@ -78,44 +70,85 @@
...
+#      set_fact:
+#        sap_hana_preconfigure_fact_minpkglist: "{{ sap_hana_preconfigure_fact_minpkglist | difference(['']) + [ pkg.stdout ] }}"
+#      loop: "{{ sap_hana_preconfigure_register_minpkglist.results }}"
+#      loop_control:
+#        loop_var: pkg
+
+  - debug: var=sap_hana_preconfigure_fact_minpkglist
+
+  - name: Install minimum packages if required
+    yum:
```
Comment (richm) [inline tasks/RedHat/installation.yml]: suggest to use the `package` module instead of `yum`

File: tasks/RedHat/installation.yml [hunk 2]
```diff
@@ -78,44 +70,85 @@
...
+      name: "{{ sap_hana_preconfigure_fact_minpkglist }}"
+      state: present
+    when: not ( sap_hana_preconfigure_fact_minpkglist == [ "" ] )
```
Comment (richm) [inline tasks/RedHat/installation.yml]: `when: sap_hana_preconfigure_fact_minpkglist | d([])`
an empty list evaluates to `false` in a boolean context, and a non-empty list evaluates to `true`.  Unless you need to explicitly check that there is a list which is not empty but contains one element which is `""`

File: tasks/RedHat/installation.yml [hunk 3]
```diff
@@ -78,44 +70,85 @@
...
 
   when: 
     - sap_hana_preconfigure_min_package_check|bool
-    - not( (sap_hana_preconfigure_min_pkgs is undefined) or (sap_hana_preconfigure_min_pkgs is none) or (sap_hana_preconfigure_min_pkgs | trim == '') ) 
+    - not( (__sap_hana_preconfigure_min_pkgs is undefined) or (__sap_hana_preconfigure_min_pkgs is none) or (__sap_hana_preconfigure_min_pkgs | trim == '') )
```
Comment (richm) [inline tasks/RedHat/installation.yml]: you could use `when: __sap_hana_preconfigure_min_pkgs | d([])`
unless you need to do various types of input validation, if the user mistakenly sets `__sap_hana_preconfigure_min_pkgs: " "`, or other user errors

Comment (berndfinger) [reply tasks/RedHat/installation.yml]: Your proposal makes sense to me.

File: tasks/RedHat/installation.yml (hunk 2 above)
Comment (berndfinger) [reply tasks/RedHat/installation.yml]: Your proposal makes sense to me.

File: tasks/RedHat/installation.yml (hunk 1 above)
Comment (berndfinger) [reply tasks/RedHat/installation.yml]: Not sure why I chose yum this time but I tested and found that the role works fine with the package module here.

**Review by richm — APPROVED** [review]
lgtm

**Pull request merged**